src/playground-backend/lib/utils.ts

```diff
@@ -15,3 +19,31 @@ export const sendJson = (
     },
   });
 };
+
+export function logError(id: string, code: string, e: Error) {
+  console.error(id + ' -- ' + new Date().toString());
+  console.error(id + ' -- ' + code);
+  console.error(id + ' -- ' + e);
+  console.error('\n\n');
+}
+
+export async function compileAskCode(
+  baseEnvironment: {
+    values?: { [p: string]: any };
+    customValues?: { [p: string]: any };
+    resources: {
+      [p: string]: Resource<any, any>;
+    };
+  },
+  askCode: AskCodeOrValue
+) {
+  let result = await runUntyped(baseEnvironment, askCode, []);
+  if (
+    result === Infinity ||
+    result === -Infinity ||
+    (typeof result === 'number' && isNaN(result))
+  ) {
+    result = result.toString();
+  }
+  return result;
+}
```

Comment:
@mhagmajer this is the "magic". Would it be valid to add this logic to the middleware as well?